lib/SPIRVProducerPass.cpp

```diff
@@ -3899,21 +3896,22 @@ void SPIRVProducerPass::GenerateInstruction(Instruction &I) {
     }
 
     if (Opcode == spv::OpPtrAccessChain) {
-      // Do we need to generate ArrayStride?  Check against the GEP result type
-      // rather than the pointer type of the base because when indexing into
-      // an OpenCL program-scope constant, we'll swap out the LLVM base pointer
-      // for something else in the SPIR-V.
-      // E.g. see test/PointerAccessChain/pointer_index_is_constant_1.cl
+      // Shader validation in the SPIR-V spec requires that the base pointer to
+      // OpPtrAccessChain (in StorageBuffer storage class) be decorated with
+      // ArrayStride.
       auto address_space = ResultType->getAddressSpace();
       setVariablePointersCapabilities(address_space);
       switch (GetStorageClass(address_space)) {
       case spv::StorageClassStorageBuffer:
-      case spv::StorageClassUniform:
```

Comment:
What's the reason for removing the `StorageClassUniform` case? AFAICT the shader validation rule isn't specific to `StorageBuffer`.

Reply:
Variables pointers only work with StorageBuffer and Workgroup storage classes and OpPtrAccessChain generates a variable pointer so clspv is generating invalid code if it does generate OpPtrAccessChain for a Uniform pointer. No point generating array stride.